src/Components/PowerBIWidget/Internal/PowerBIWidgetBuilder/PowerBIWidgetBuilder.tsx

```diff
@@ -0,0 +1,288 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import produce from 'immer';
+import {
+    IPowerBIWidgetBuilderProps,
+    IPowerBIWidgetBuilderStyleProps,
+    IPowerBIWidgetBuilderStyles
+} from './PowerBIWidgetBuilder.types';
+import { getStyles } from './PowerBIWidgetBuilder.styles';
+import {
+    classNamesFunction,
+    useTheme,
+    styled,
+    Stack,
+    TextField,
+    ComboBox
+} from '@fluentui/react';
+import { useTranslation } from 'react-i18next';
+import { getWidgetFormStyles } from '../../../ADT3DSceneBuilder/Internal/Behaviors/Widgets/WidgetForm/WidgetForm.styles';
+
+const getClassNames = classNamesFunction<
+    IPowerBIWidgetBuilderStyleProps,
+    IPowerBIWidgetBuilderStyles
+>();
+
+const PowerBIWidgetBuilder: React.FC<IPowerBIWidgetBuilderProps> = ({
+    styles,
+    formData,
+    adapter,
+    updateWidgetData,
+    setIsWidgetConfigValid
+}) => {
+    // contexts
+
+    // state
+    const [selectedType, setSelectedType] = useState(
```

Comment:
does this automatically know the type is not just string but the specific enum you defined in config? if not, can we make it explicit to avoid mistakes of setting a rogue string to this state?

Reply:
It does. It becomes `const setSelectedType: React.Dispatch<React.SetStateAction<"Visual" | "Tile">>`